File: hw/quiroga/U2/Hw18Exceptions/Inheritance_JS/javascript/ec/edu/espe/zoo/model/Bird.mjs
import { Animal } from "./Animal.mjs";


export class Bird extends Animal {
    #wingSize;
    #canFly;

    constructor(wingSize, canFly, id, name, bornOnDate, weight, sex, vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.#wingSize = wingSize;
        this.#canFly = canFly;
    }

    chirp() {
        console.log(`${super.getName()} is chirping`);
    }

    migrate(location) {
        console.log(`${super.getName()} is migrating to ${location}`);
    }

    toString() {
        return `Bird{${super.toString()} wingSize=${this.#wingSize}, canFly=${this.#canFly}}`;
    }

    getWingSize() {
        return this.#wingSize;
    }

    setWingSize(wingSize) {
        this.#wingSize = wingSize;
    }

    isCanFly() {
        return this.#canFly;
    }

    setCanFly(canFly) {
        this.#canFly = canFly;
    }
}

File: u2/hw18Inheritance/java/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Bird.java


package ec.edu.espe.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Robinson Bonilla
 */
public class Bird extends Animal{
    private float wingSize;
    private boolean canFly;
    
    public void chirp(){
        System.out.println(super.getName()+" is chirping");
    }
    public void migrate(String location){
        System.out.println(super.getName()+" is migrating to"+location);
    }
    @Override
    public String toString() {
        return "Bird{" +super.toString()+ "wingSize=" + wingSize + ", canFly=" + canFly + '}';
    }
    
    

    public Bird(float wingSize, boolean canFly, int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.wingSize = wingSize;
        this.canFly = canFly;
    }
    
    

    /**
     * @return the wingSize
     */
    public float getWingSize() {
        return wingSize;
    }

    /**
     * @param wingSize the wingSize to set
     */
    public void setWingSize(float wingSize) {
        this.wingSize = wingSize;
    }

    /**
     * @return the canFly
     */
    public boolean isCanFly() {
        return canFly;
    }

    /**
     * @param canFly the canFly to set
     */
    public void setCanFly(boolean canFly) {
        this.canFly = canFly;
    }
    

}
